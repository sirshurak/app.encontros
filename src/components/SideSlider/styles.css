#side-slider .swiper-slide {
  font-size: 18px;
  /* Center slide text vertically */
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-align-items: center;
  align-items: center;
  width: 100px !important;
  margin: 0 15px;
}

#side-slider .swiper-slide > a {
  display: flex;
}

#side-slider .swiper-scrollbar {
  background-color: inherit;
}

#side-slider .swiper-scrollbar-drag {
  background-color: #5a6570;
  opacity: 0.2;
}

#side-slider h2 {
  text-align: left;
  margin: 0px 0px 0px 15px;
}

