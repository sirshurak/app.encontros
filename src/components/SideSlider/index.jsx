import React from "react";
import SwiperCore, { Navigation, Pagination, Scrollbar, A11y } from "swiper";
import { Swiper, SwiperSlide } from "swiper/react";

// Import Swiper styles
import "swiper/swiper.scss";
import "swiper/components/navigation/navigation.scss";
import "swiper/components/pagination/pagination.scss";
import "swiper/components/scrollbar/scrollbar.scss";

import "./styles.css";

// install Swiper components
SwiperCore.use([Navigation, Pagination, Scrollbar, A11y]);

function SideSlider({ title, children }) {
  return (
    <section id="side-slider">
      <h2>{title}</h2>
      <Swiper
        slidesPerView="auto"
        scrollbar={{ draggable: true, hide: true }}
        observer
        observeParents
      >
        {children?.length &&
          children.map((child, key) => (
            <SwiperSlide key={key}>{child}</SwiperSlide>
          ))}
      </Swiper>
    </section>
  );
}

export default SideSlider;
