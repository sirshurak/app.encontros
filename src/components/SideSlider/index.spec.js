import React from 'react';
import { shallow, mount } from 'enzyme';
import SideSlider from './index';

const SliderItems = 
        [
            (<span className="item-1" key={1}>item 1</span>),
            (<span className="item-2" key={2}>item 2</span>),
            (<span className="item-3" key={3}>item 3</span>)
        ];

describe('SideSlider component', () => {
    it('should match shallow snapshop', () => {
        const component = shallow(<SideSlider title="Title">{SliderItems}</SideSlider>)
        expect(component).toMatchSnapshot();
    });

    it('should mount slider items', () => {
        const component = mount(<SideSlider title="Title">{SliderItems}</SideSlider>);

        expect(component.find('h2').text()).toBe('Title');
        expect(component.find('.item-1')).toExist();
        expect(component.find('.item-2')).toExist();
        expect(component.find('.item-3')).toExist();
        expect(component.find('.item-4')).not.toExist();
    });
});