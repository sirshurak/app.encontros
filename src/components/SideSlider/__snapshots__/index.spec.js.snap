// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideSlider component should match shallow snapshop 1`] = `
<section
  id="side-slider"
>
  <h2>
    Title
  </h2>
  <Swiper
    observeParents={true}
    observer={true}
    scrollbar={
      Object {
        "draggable": true,
        "hide": true,
      }
    }
    slidesPerView="auto"
  >
    <SwiperSlide
      key="0"
    >
      <span
        className="item-1"
        key="1"
      >
        item 1
      </span>
    </SwiperSlide>
    <SwiperSlide
      key="1"
    >
      <span
        className="item-2"
        key="2"
      >
        item 2
      </span>
    </SwiperSlide>
    <SwiperSlide
      key="2"
    >
      <span
        className="item-3"
        key="3"
      >
        item 3
      </span>
    </SwiperSlide>
  </Swiper>
</section>
`;
